addons/payment_test/controllers/main.py

```diff
@@ -1,26 +1,57 @@
 
-from odoo import http, _
+import logging
+from random import randint
+
+from odoo import http
 from odoo.http import request
-from odoo.exceptions import UserError
+
+from odoo.addons.payment import utils as payment_utils
+
+_logger = logging.getLogger(__name__)
 
 
 class PaymentTestController(http.Controller):
 
-    @http.route(['/payment/test/s2s/create_json_3ds'], type='json', auth='public', csrf=False)
-    def payment_test_s2s_create_json_3ds(self, verify_validity=False, **kwargs):
-        if not kwargs.get('partner_id'):
-            kwargs = dict(kwargs, partner_id=request.env.user.partner_id.id)
-        acquirer = request.env['payment.acquirer'].browse(int(kwargs.get('acquirer_id')))
-        if acquirer.state != 'test':
-            raise UserError(_("Please do not use this acquirer for a production environment!"))
-        token = acquirer.s2s_process(kwargs)
-
-        return {
-            'result': True,
-            'id': token.id,
-            'short_name': 'short_name',
+    @http.route('/payment/test/payment', type='json', auth='public')
+    def process_payment(self, reference, cc_number, cc_cvc, cc_name, cc_expiry):
+        """ Simulate the result of a payment request and handle the response.
+
+        :param str reference: The reference of the transaction
+        :param int cc_number: The credit card number
+        :param int cc_cvc: The credit card cvc
+        :param int cc_name: the credit card owner
+        :param int cc_expiry: The credit card expiry date
+        :return: The JSON-formatted content of the response
+        :rtype: dict
+        """
+        fake_api_response = {
+            'status': 'success',
             '3d_secure': False,
             'verified': True,
+            'acquirer_reference': randint(1, 10000),
+            'cc_number': payment_utils.build_token_name(payment_details_short=cc_number[-4:]),
+            'cc_cvc': 'XXX',
+            'cc_expiry': cc_expiry,
+            'cc_holder_name': cc_name,
         }
+        # Handle the payment
+        tx_sudo = request.env['payment.transaction'].sudo().search([('reference', '=', reference)])
+        values = {'acquirer_reference': fake_api_response.get('acquirer_reference')}
+        # Create the token with the data of the payment method
+        if tx_sudo.tokenize:
+            token_vals = {
+                'name': f"TEST {fake_api_response['cc_number']} "
+                        f"{fake_api_response['cc_holder_name']}",
+                'partner_id': tx_sudo.partner_id.id,
+                'acquirer_id': tx_sudo.acquirer_id.id,
+                'acquirer_ref': fake_api_response['acquirer_reference'],
+                'verified': True,
+            }
+            token = request.env['payment.token'].create(token_vals)
+            values['token_id'] = token.id
+        tx_sudo.write(values)
+        tx_sudo._set_done()
+        tx_sudo._execute_callback()
+        return {}
```

Comment:
I may have told you otherwise, apologies if it's the case, but now that I'm seeing this I'm convinced that the handling of the fake API response should be done in `_handle_feedback_data` as usual.
This would:
1. Limit the role of this route to faking an API response thus making it cleaner, and allowing to use it for other (future) purposes.
2. Strictly emulate the payment flows which is what we want.
3. Allow overrides to apply to `payment_test` too.
4. Make the code more resilient: `_execute_callback`, for example, should ideally always be called from `_handle_feedback_data` when possible.